internal/configsource/fileconfigsource/session_test.go

```diff
@@ -0,0 +1,177 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fileconfigsource
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/config/experimental/configsource"
+)
+
+func TestFileConfigSource_Session(t *testing.T) {
+	tests := []struct {
+		defaults map[string]interface{}
+		params   map[string]interface{}
+		expected interface{}
+		wantErr  error
+		name     string
+		selector string
+	}{
+		{
+			name:     "simple",
+			selector: "scalar_data_file",
+			expected: []byte("42"),
+		},
+		{
+			name: "invalid_param",
+			params: map[string]interface{}{
+				"unknown_params_field": true,
+			},
+			wantErr: &errInvalidRetrieveParams{},
+		},
+		{
+			name:     "missing_file",
+			selector: "not_to_be_found",
+			wantErr:  &errMissingRequiredFile{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s, err := newSession()
+			require.NoError(t, err)
+			require.NotNil(t, s)
+
+			ctx := context.Background()
+			defer func() {
+				assert.NoError(t, s.RetrieveEnd(ctx))
+				assert.NoError(t, s.Close(ctx))
+			}()
+
+			file := path.Join("testdata", tt.selector)
+			r, err := s.Retrieve(ctx, file, tt.params)
+			if tt.wantErr != nil {
+				assert.Nil(t, r)
+				require.IsType(t, tt.wantErr, err)
+				return
+			}
+
+			require.NoError(t, err)
+			require.NotNil(t, r)
+			assert.Equal(t, tt.expected, r.Value())
+		})
+	}
+}
+
+func TestFileConfigSource_DeleteFile(t *testing.T) {
+	s, err := newSession()
+	require.NoError(t, err)
+	require.NotNil(t, s)
+
+	ctx := context.Background()
+	defer func() {
+		assert.NoError(t, s.RetrieveEnd(ctx))
+		assert.NoError(t, s.Close(ctx))
+	}()
+
+	// Copy test file
+	src := path.Join("testdata", "scalar_data_file")
+	contents, err := ioutil.ReadFile(src)
+	require.NoError(t, err)
+	dst := path.Join("testdata", "copy_scalar_data_file")
+	require.NoError(t, ioutil.WriteFile(dst, contents, 0644))
+	t.Cleanup(func() {
+		// It should be removed prior to this so an error is expected.
+		assert.Error(t, os.Remove(dst))
+	})
+
+	params := map[string]interface{}{
+		"delete": true,
+	}
+	r, err := s.Retrieve(ctx, dst, params)
+
+	require.NoError(t, err)
+	require.NotNil(t, r)
+	assert.Equal(t, []byte("42"), r.Value())
+
+	assert.Equal(t, configsource.ErrWatcherNotSupported, r.WatchForUpdate())
+}
+
+func TestFileConfigSource_DeleteFileError(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		t.Skip("Windows only test")
```

Comment:
It's not jumping out to me why this is the case, so could the title be updated or a comment provided?  May be overkill, but adding a `session_windows_test.go` file w/ windows build tag could also be a helpful start.

Reply:
Added comment to explain running only on Windows: locking the file is trivial on Windows, but not on *nix given the golang API (require lower-level non-portable calls). For now, I will not add the separate file but if there is time to enable the test on *nix I will separate the files with the "lock" function.